### PR TITLE
Fix creation of views with comments using `databricks_sql_table`

### LINKED ISSUE #2573: [ISSUE] Issue with `databricks_sql_table` resource with `table_type` is VIEW
### Configuration
```hcl
#example.yaml
storage_acccount_name: "storageaccountname"
storage_container_name: "containername"
capture_id: "test"
type: "type123"
file_format: "PARQUET"
table_definition:
  table_name: "bronze_example"
  table_description: "This is an example table view created with Terraform"
  read_access: ["team-data"]
  sql_definition: "SELECT time, get_json_object(data, '$.deviceProfile.customKeyboards') as device_profile_custom_keyboards, get_json_object(data, '$.type') as type FROM %s WHERE type == 'type123"
  columns: {device_profile_custom_keyboards: {type: "string", comment: "Dummy column"}}

# locals.tf
locals{
integration_configuration = {
    example = yamldecode(file("./example.yaml"))
  }
}

# main.tf
resource "databricks_sql_table" "example_bronze" {
  provider        = databricks.admin_workspace
  name            = local.integration_configuration.example.table_definition.table_name
  catalog_name    = databricks_catalog.integration_events.name
  schema_name     = databricks_schema.raw.name
  table_type      = "VIEW"
  view_definition = format(local.integration_configuration.example.table_definition.sql_definition, databricks_sql_table.example_raw.id)

  dynamic "column" {
    for_each = tomap(local.integration_configuration.example.table_definition.columns)
    content {
      name    = column.key
      type    = column.value.type
      comment = column.value.comment
    }
  }
  comment = local.integration_configuration.example.table_definition.table_description
}

```

### Expected Behavior
A view table is created with columns given in `view_definition` and metadata set in the `column` block are enforced.

### Actual Behavior
The following error is given upon `terraform apply`:
````
│ Error: cannot create sql table: cannot execute CREATE VIEW integration_events.raw.bronze_example (device_profile_custom_keyboards string COMMENT 'Dummy column')
│ COMMENT 'This is an example table view created with Terraform'
│ AS SELECT time, get_json_object(data, '$.deviceProfile.customKeyboards') as device_profile_custom_keyboards, get_json_object(data, '$.type') as type FROM integration_events.raw.example WHERE type == 'type123';: [PARSE_SYNTAX_ERROR] Syntax error at or near 'string'.(line 1, pos 83)
│ 
│   with databricks_sql_table.example_bronze,
│   on integrations.tf line 64, in resource "databricks_sql_table" "example_bronze":
│   64: resource "databricks_sql_table" "example_bronze" {
````
### Steps to Reproduce
`terraform apply`

### Terraform and provider versions
Terraform: 1.5.4
Databricks: 1.22.0

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
Fix creation of views with comments using `databricks_sql_table`. Per [doc](https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-view.html), column type is not needed of views so marking it as optional

Closes #2573

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Comment by nkvuong** [comment]
need to suppress diff as UC server returns the column type

File: docs/resources/sql_table.md [hunk 1]
```diff
@@ -82,11 +82,12 @@ The following arguments are supported:
 * `properties` - (Optional) Extensible Table properties.
 
 ### `column` configuration block
+
 For table columns
 Currently, changing the column definitions for a table will require dropping and re-creating the table
 
 * `name` - User-visible name of column
-* `type` - Column type spec (with metadata) as SQL text
+* `type` - (Optional for `VIEW`) Column type spec (with metadata) as SQL text
```
Comment (mgyucht) [inline docs/resources/sql_table.md:90]: Is it optional, or is it invalid to specify with view?

**Review by mgyucht — APPROVED** [review]
LGTM, just one docs question.

File: docs/resources/sql_table.md (hunk 1 above)
Comment (nkvuong) [reply docs/resources/sql_table.md]: good point, I have updated the doc

**Pull request merged**